File: templates/problem-template.js
/**
 * NeetCode 150 - JavaScript Problem Template
 * Run: node solution.js
 */

class Solution {
    // TODO: Implement your solution here
    
}

// ==================== TEST HARNESS ====================

function runTest(testName, actual, expected) {
    const pass = JSON.stringify(actual) === JSON.stringify(expected);
    console.log(`Test (${testName}): ${pass ? 'PASS' : 'FAIL'}`);
    if (!pass) {
        console.log(`  Expected: ${JSON.stringify(expected)}`);
        console.log(`  Actual:   ${JSON.stringify(actual)}`);
    }
    return pass;
}

function main() {
    const sol = new Solution();
    let allPassed = true;

    // Example usage (replace with actual problem tests):
    // allPassed = runTest("Example 1", sol.yourMethod([1, 2, 3]), 6) && allPassed;
    // allPassed = runTest("Example 2", sol.yourMethod([]), 0) && allPassed;

    console.log(allPassed ? "\nAll tests passed!" : "\nSome tests failed.");
    process.exit(allPassed ? 0 : 1);
}

main();

File: templates/problem-template.cs
/**
 * NeetCode 150 - C# Problem Template
 * Run: dotnet run problem.cs
 * Or: csc problem.cs && mono problem.exe
 */

using System;
using System.Collections.Generic;

public class Solution {
    public bool IsAnagram(string s, string t) {
        // TODO: Implement your solution here
        return false;
    }
}

// ==================== TEST HARNESS ====================

public class Program {
    private static int testNum = 0;
    private static bool allPassed = true;

    private static void RunTest(string testName, string s, string t, bool expected) {
        testNum++;
        Solution sol = new Solution();
        bool actual = sol.IsAnagram(s, t);
        bool pass = (actual == expected);
        if (!pass) allPassed = false;

        Console.WriteLine($"Test {testNum} ({testName}): {(pass ? "PASS" : "FAIL")}");
        if (!pass) {
            Console.WriteLine($"  s: \"{s}\", t: \"{t}\"");
            Console.WriteLine($"  Expected: {expected}");
            Console.WriteLine($"  Actual:   {actual}");
        }
    }

    public static void Main(string[] args) {
        // Example usage (replace with actual problem tests):
        // RunTest("Example 1", "anagram", "nagaram", true);

        Console.WriteLine(allPassed ? "\nAll tests passed!" : "\nSome tests failed.");
        Environment.Exit(allPassed ? 0 : 1);
    }
}
